src/pages/Editor/containers/editContext.js

```diff
@@ -11,10 +11,69 @@ const svgStyles = {
   top: 0,
   left: 0,
 };
-const EditContextProvider = props => {
+const EditContextProvider = (props) => {
   const aImagePrefix = "";
   const [pageSrc, setPageSrc] = useState(`${aImagePrefix}blank1.png`);
   const [isBody, setIsBody] = useState(true);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [pages, setPages] = useState([
+    {
+      head: {
+        text: "",
+        headLeft: 0,
+        headTop: 0,
+        headWidth: null,
+      },
+      body: {
+        text: "",
+        bodyLeft: 0,
+        bodyTop: 0,
+        bodyWidth: null,
+      },
+    },
+  ]);
+
+  useEffect(() => {
+    if (currentPage === pages.length) {
+      setCurrentPage((currentPage) => currentPage - 1);
+    }
+  });
+
+  const deletePage = (index) => {
+    var pagesCopy = [...pages];
+    pagesCopy.splice(index, 1);
+    setPages(pagesCopy);
+  };
+  const addPage = () => {
+    var pagesCopy = [...pages];
+    pagesCopy.push({
+      head: {
+        text:"",
+        headLeft: 0,
+        headTop: 0,
+        headWidth: null,
+      },
+      body: {
+        text:"",
+        bodyLeft: 0,
+        bodyTop: 0,
+        bodyWidth: null,
+      },
+    });
+    setPages(pagesCopy);
+  };
+
+  const nextPage = () => {
+    if (currentPage < pages.length - 1) {
+      setCurrentPage((currentPage) => currentPage + 1);
+    }
+  };
```

Comment:
```suggestion
  const nextPage = () => currentPage < pages.length - 1 && setCurrentPage((currentPage) => currentPage + 1);
```
Better to do it this way.